app/src/main/kotlin/com/wire/android/ui/main/MainDrawer.kt

```diff
@@ -0,0 +1,159 @@
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import com.wire.android.R
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@Composable
+fun Drawer(scope: CoroutineScope, scaffoldState: ScaffoldState, navController: NavController) {
+
+    val topItems = listOf(MainScreen.Conversations, MainScreen.Archive, MainScreen.Vault)
+    val bottomItems = listOf(MainScreen.Settings, MainScreen.Support)
+
+    Column(
+        modifier = Modifier
+            .padding(top = 40.dp, start = 8.dp, end = 8.dp, bottom = 16.dp)
+
+    ) {
+        Image(
+            painter = painterResource(id = R.drawable.ic_big_logo),
+            contentDescription = "logo",
+            contentScale = ContentScale.Fit,
+            modifier = Modifier
+                .width(80.dp)
+                .height(30.dp)
+                .padding(8.dp)
+        )
+
+        Spacer(
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(16.dp)
+        )
+
+        val navBackStackEntry by navController.currentBackStackEntryAsState()
+        val currentRoute = navBackStackEntry?.destination?.route
+
+        topItems.forEach { item ->
+            DrawerItem(item = item,
+                selected = currentRoute == item.route,
+                onItemClick = {
+                    itemClickActions(navController, item, scope, scaffoldState)
+                })
+        }
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        bottomItems.forEach { item ->
+            DrawerItem(item = item,
+                selected = currentRoute == item.route,
+                onItemClick = {
+                    itemClickActions(navController, item, scope, scaffoldState)
+                })
+        }
+
+    }
+}
+
+private fun itemClickActions(
+    navController: NavController,
+    item: MainScreen,
+    scope: CoroutineScope,
+    scaffoldState: ScaffoldState
+) {
+    navController.navigate(item.route) {
+        navController.graph.startDestinationRoute?.let { route ->
+            popUpTo(route) {
+                saveState = true
+            }
+        }
+        launchSingleTop = true
+        restoreState = true
+    }
+
+    scope.launch { scaffoldState.drawerState.close() }
+}
+
+@Composable
+fun DrawerItem(item: MainScreen, selected: Boolean, onItemClick: (MainScreen) -> Unit) {
+    val backgroundColor = if (selected) colorResource(id = R.color.light_blue) else colorResource(id = android.R.color.transparent)
+    val contentColor = if (selected) colorResource(id = R.color.text_white) else colorResource(id = R.color.text_black)
```

Comment:
I think the right place for Color is under theme>Color.kt I think when you worked on it, it was not on develop like this yet :S I merged it to develop now however, also we agreed to develop in a way to support dark mode in the future, maybe take a look at my PR https://github.com/wireapp/wire-android-reloaded/pull/303

Reply:
The following link offers a nice resume on how to deal with colors and themes in Compose 
https://developer.android.com/jetpack/compose/themes/material